doc/langdef.md

```diff
@@ -1208,1337 +1207,938 @@ Regular expressions follow the
+*   `bytes <= bytes -> bool`
+*   `google.protobuf.Timestamp <= google.protobuf.Timestamp -> bool`
+*   `google.protobuf.Duration <= google.protobuf.Duration -> bool`
+
+**Examples:**
+
+```
+2 <= 3 // true
+'a' <= 'b' // true
+timestamp('2023-08-25T12:00:00Z') <= timestamp('2023-08-26T12:00:00Z') // true
+```
+
+**Less Than (\<)** \- Compares two values and returns `true` if the first value
+is less than the second value, and `false` otherwise
+
+**Signatures:**
+
+*   `bool < bool -> bool`
+*   `int < int -> bool`
+*   `uint < uint -> bool`
+*   `double < double -> bool`
+*   `string < string -> bool`
+*   `bytes < bytes -> bool`
+*   `google.protobuf.Timestamp < google.protobuf.Timestamp -> bool`
+*   `google.protobuf.Duration < google.protobuf.Duration -> bool`
+
+**Examples:**
+
+```
+2 < 3 // true
+'a' < 'b' // true
+duration('2h') < duration('3h') // true
+-1 < dyn(1u) // true
+```
+
+**Greater Than or Equal To (\>=)** \- Compares two values and returns `true` if
+the first value is greater than or equal to the second value, and `false`
+otherwise
+
+**Signatures:**
+
+*   `bool >= bool -> bool`
+*   `int >= int -> bool`
+*   `uint >= uint -> bool`
+*   `double >= double -> bool`
+*   `string >= string -> bool`
+*   `bytes >= bytes -> bool`
+*   `google.protobuf.Timestamp >= google.protobuf.Timestamp -> bool`
+*   `google.protobuf.Duration >= google.protobuf.Duration -> bool`
+
+**Examples:**
+
+```
+3 >= 2 // true
+'b' >= 'a' // true
+duration('2h') + duration('1h1m') >= duration('3h') // true
+1 >= dyn(18446744073709551615u) // false
+```
+
+**Greater Than (\>)** \- Compares two values and returns `true` if the first
+value is greater than the second value, and `false` otherwise
+
+**Signatures:**
+
+*   `bool > bool -> bool`
+*   `int > int -> bool`
+*   `uint > uint -> bool`
+*   `double > double -> bool`
+*   `string > string -> bool`
+*   `bytes > bytes -> bool`
+*   `google.protobuf.Timestamp > google.protobuf.Timestamp -> bool`
+*   `google.protobuf.Duration > google.protobuf.Duration -> bool`
+
+**Examples:**
+
+```
+3 > 2 // true
+'b' > 'a' // true
+5u > 3u // true
+```
+
+#### List Operators
+
+**List Indexing (\[\])** \- list indexing. Constant time cost
+
+**Signatures:**
+
+*   `list(A)[int] -> A`
+
+**Examples:**
+
+```
+[1, 2, 3][1] // 2
+```
+
+**List Membership (in)** \- Checks if a value is present in a list. Time cost is
+proportional to the product of the size of both arguments.
+
+**Signatures:**
+
+*   `A in list(A) -> bool`
+
+**Examples:**
+
+```
+2 in [1, 2, 3] // true
+"a" in ["b", "c"] // false
+```
+
+**size** \- Determine the number of elements in the list.
+
+**Signatures:**
+
+*   `list.size() -> int`
+*   `size(list) -> int`
+
+**Examples:**
+
+```
+['hello', 'world'].size() // 2
+size(['first', 'second', 'third']) // 3
+```
+
+#### Map Operators
+
+**Map Indexing (\[\])** \- map indexing. For string keys, cost is proportional
+to the size of the map keys times the size of the index string
```

Comment:
Up to you: this is technically correct, but practically this is always a hash map so maybe worth suggesting expected O(1) in practice as with the `in` description?